v2/dtos/profileresource.go

```diff
@@ -0,0 +1,35 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package dtos
+
+import "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+
+// ProfileResource defines read/write capabilities native to the device
+// This object and its properties correspond to the ProfileResource object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/ProfileResource
+type ProfileResource struct {
+	Name string              `json:"name,omitempty" yaml:"name,omitempty"`
```

Comment:
Name should be required since it is used in a Add request and Get and Put can't both be empty??

Reply:
Added tags:
* name:  validate:"required"
* Get:  validate:"required_without=Set"
* Set:  validate:"required_without=Get"